internal/bminventory/inventory.go

```diff
@@ -1658,6 +1659,26 @@ func (b *bareMetalInventory) updateHostsData(ctx context.Context, params install
 		}
 	}
 
+	for i := range params.ClusterUpdateParams.HostsMachineConfigPoolNames {
```

Comment:
maybe it should be a helper function? looks like this function can be split

Reply:
This function would anyhow get changed for [MGMT-3129 Merge updatable host fields on cluster-update-params](https://issues.redhat.com/browse/MGMT-3129)

Reply:
Until it happens let's not make the code worse